dev/.scalafmt.conf

```diff
@@ -0,0 +1,24 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+align = none
+align.openParenDefnSite = false
+align.openParenCallSite = false
+align.tokens = []
+docstrings = JavaDoc
+maxColumn = 98
```

Comment:
Are we using 100 for maxColumn?

Reply:
I originally had this set to 100, and noticed a corner case where an ending space and curly brace were formatted on columns 100 and 101 respectively, making scalastyle complain.

Reply:
If we set it as `98`, will this complain for legit code with 100 chars?

Reply:
For now, this won't complain about any code, it just gives a way for contributors to format it.

If we did turn on test as part of the build process, it would complain about any code that was different after formatting.  That could include code in that 99-101 column range, depending on whether it hit that corner case.  E.g.

```
    // this is 101 columns and will fail scalastyle, but will pass with scalafmt set to 99
    if (caseInsensitiveParams.contains(s"kafka.${ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG}")) {
```

My thinking is that being slightly more strict than existing scalastyle is better than having automated formatting that won't pass other existing automated checks.  Trying to find and upstream fixes to off-by-one errors in scalastyle is another option, but that will take a while.